ctapipe/ml/stereo_combination.py

```diff
@@ -0,0 +1,141 @@
+from abc import abstractmethod
+import numpy as np
+from astropy.table import Table
+import astropy.units as u
+from scipy.ndimage import median
+from ctapipe.core import Component
+from ctapipe.ml.preprocessing import check_valid_rows
+
+
+def _calculate_ufunc_of_telescope_values(tel_data, n_array_events, indices, ufunc):
+    mean_values = np.zeros(n_array_events)
+    ufunc.at(mean_values, indices, tel_data)
+    return mean_values
+
+
+class StereoCombiner(Component):
+    # TODO: Add quality query (after #1888)
+    def __init__(self, mono_prediction_column, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.mono_prediction_column = mono_prediction_column
+
+    @abstractmethod
+    def __call__(self, mono_predictions: Table) -> np.ndarray:
+        """
+        Constructs stereo predictions from a table of
+        telescope events.
+        """
+
+    def check_valid(self, table: Table) -> np.ndarray:
+        """
+        Selects the telescope events, that should be used
+        for the stereo reconstruction.
+        Due to the quality query only operating on single events currently,
+        as of now this only discards rows with nans in them.
+        """
+        valid = check_valid_rows(table)
+        # TODO: Add quality query (after #1888)
+        return valid
+
+
+class StereoMeanCombiner(StereoCombiner):
+    """
+    Calculates array-wide (stereo) predictions as the mean of
+    the reconstruction on telescope-level with an optional weighting.
+    """
+
+    def __init__(self, mono_prediction_column, weight_column=None, *args, **kwargs):
+        super().__init__(mono_prediction_column=mono_prediction_column, *args, **kwargs)
+        self.weight_column = weight_column
```

Comment:
We won't have a weight column... I think we should implement an enum of weights, e.g. like this:

```
weight = Enum(["intensity", "konrad"])
```

where "intensity" just means weighting by intensity and "konrad" is the definition also used in `HillasReconstructor`. The weight should be calculated here.